Mismatched Types on Master Branch

Hi all,
As with the latest issues with the Pi 4 that require using the master branch, I have switched over accordingly (I believe that was clarified in issue #73 or #75.) However, when I attempt to build any project with `rppal` as of the last week, cargo exits with an error.

Steps to reproduce:
- Require `rppal` from git (specifically, I did `rppal = { git = "https://github.com/golemparts/rppal.git" }`)
- Attempt to run/build/check with cargo (essentially try to compile)
- Errors with this message:
```rust
error[E0308]: mismatched types
  --> /home/danya/.local/share/cargo/git/checkouts/rppal-451d1b4d0438fc03/a1ab827/src/pwm/sysfs.rs:59:17
   |
59 |                 buf.as_mut_ptr(),
   |                 ^^^^^^^^^^^^^^^^ expected `i8`, found `u8`
   |
   = note: expected raw pointer `*mut i8`
              found raw pointer `*mut u8`
```

This has occurred on all projects, from simple to complex, across many computers, so I don't believe it is simply an issue of my setup. This has popped up within the past week, but I hadn't seen any commits since February. I know this issue was to have been fixed in version `0.12.0`, any update on that?

As a very unsafe monkey patch, I've cast the issue on lines 59 and 89 to `as *mut i8` and it does work in my application, but that definitely isn't a permanent fix. 

I'll provide whatever else I can if needed.

--- a/src/pwm/sysfs.rs
+++ b/src/pwm/sysfs.rs
@@ -75,7 +75,7 @@ pub fn user_to_uid(name: &str) -> Option<u32> {
 // Find group ID for specified group
 pub fn group_to_gid(name: &str) -> Option<u32> {
     if let Ok(name_cstr) = CString::new(name) {
-        let buf = &mut [0_u8; 4096];
+        let buf = &mut [0 as c_char; 4096];
         let mut res: *mut group = ptr::null_mut();
         let mut grp = group {
             gr_name: ptr::null_mut(),

